test/functional/p2p_addr_relay.py

```diff
@@ -82,6 +122,95 @@ def run_test(self):
         ipv4_branching_factor = 2
         assert_equal(total_ipv4_received, num_ipv4_addrs * ipv4_branching_factor)
 
+        self.nodes[0].disconnect_p2ps()
+
+        self.log.info('Check relay of addresses received from outbound peers')
+        inbound_peer = self.nodes[0].add_p2p_connection(AddrReceiver())
+        full_outbound_peer = self.nodes[0].add_outbound_p2p_connection(GetAddrStore(), p2p_idx=0, connection_type="outbound-full-relay")
+        msg = self.setup_addr_msg(2)
+        self.send_addr_msg(full_outbound_peer, msg, [inbound_peer])
+        self.log.info('Check that the first addr message received from an outbound peer is not relayed')
+        # Currently, there is a flag that prevents the first addr message received
+        # from a new outbound peer to be relayed to others. Originally meant to prevent
+        # large GETADDR responses from being relayed, it now typically affects the self-announcement
+        # of the outbound peer which is often sent before the GETADDR response.
+        assert_equal(inbound_peer.num_ipv4_received, 0)
+
+        self.log.info('Check that subsequent addr messages sent from an outbound peer are relayed')
+        msg2 = self.setup_addr_msg(2)
+        self.send_addr_msg(full_outbound_peer, msg2, [inbound_peer])
+        assert_equal(inbound_peer.num_ipv4_received, 2)
```

Comment:
This fails:

* https://cirrus-ci.com/task/4959397244829696?logs=ci#L6318

Reply:
The expected addr message is sent directly after the assert has failed, so it looks like a timing issue:
` node0 2021-04-27T06:55:10.562357Z (mocktime: 2021-04-27T07:10:08Z) [net.cpp:2967] [PushMessage] sending addr (61 bytes) peer=9 ` (https://cirrus-ci.com/task/4959397244829696?logs=ci#L6320)
With `AVG_ADDRESS_BROADCAST_INTERVAL=30s`, I thought that a mocktime of 5 mins should be plenty to prevent Poisson timer fails - I think it's not that.
Possibly, there needs to be an additional wait after the mocktime to give the node more time to actually send out the addr message?

I'll look into this more closely later today!

Reply:
fixed in #21785